subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -203,8 +216,9 @@ private TransientConfigurationResults loadTransientGraphResults(SelectedArtifact
      * Recursive but excludes unsuccessfully resolved artifacts.
      */
     public Set<File> getFiles(Spec<? super Dependency> dependencySpec) {
-        FilesAndArtifactCollectingVisitor visitor = new FilesAndArtifactCollectingVisitor();
-        visitArtifacts(dependencySpec, getSelectedArtifacts(), getSelectedFiles(), visitor);
+        LenientFilesAndArtifactResolveVisitor visitor = new LenientFilesAndArtifactResolveVisitor();
+        visitArtifactsWithBuildOperation(dependencySpec, getSelectedArtifacts(), getSelectedFiles(), visitor, null);
+        visitor.addArtifactsFiles();
```

Comment:
:x: This should not be necessary. Instead, overwrite visitor.getFiles() to do this transparently.